### PR TITLE
Revert to put widget_class in the data portion of comm message

### LINKED ISSUE #522: widget_class information on comm_open should be placed in data
The current code merged in #461 places the information about `widget_class` in a the `meta-data` section of the `comm_open` message. This is a huge deal, because nowhere in the spec around Comm support (and I'm basing the statement on [this](http://jupyter-client.readthedocs.org/en/latest/messaging.html#opening-a-comm)) does it say that handlers to comm message (open, msg, close, info) need access to the `meta-data` area. That being the case, both the Comm support in IRKernel and Toree/SparkKernel, do not make this area of the message available to handlers. In my opinion, the goal of Comm is to simplify communication and not to expose all the pieces of the Kernel Protocol.

I propose that this is changed so that the information about `widget_class` is passed within the `data` area, a much easier fix for declarativewidgets and do not involve changes to the kernels.

/cc @SylvainCorlay @jdfreder @jasongrout @parente

### PR DESCRIPTION
**Pull request by lbustelo**
Resolves #522

### CODE REVIEW

**Comment by lbustelo** [comment]
@SylvainCorlay and @jasongrout I think this is what you guys meant with keeping it symmetric. Both sides are now using the `data` portion of the comm message to share `widget-class` during the `comm_open`.

File: widgetsnbextension/src/manager.js [hunk 1]
```diff
@@ -593,12 +593,12 @@ WidgetManager.prototype.deleteSnapshots = function() {
...
-WidgetManager.prototype._create_comm = function(comm_target_name, model_id, metadata) {
+WidgetManager.prototype._create_comm = function(comm_target_name, model_id, data) {
     var that = this;
     return this._get_connected_kernel().then(function(kernel) {
-        if (metadata) {
-            return kernel.comm_manager.new_comm(comm_target_name, {},
-                                                that.callbacks(), metadata, model_id);
+        if (data) {
+            return kernel.comm_manager.new_comm(comm_target_name, data,
+                                                that.callbacks(), {}, model_id);
```
Comment (SylvainCorlay) [inline widgetsnbextension/src/manager.js]: From a jupyter-js-widgets perspective, this is a backward incompatible change.

Could we do 
```
return kernel.comm_manager.new_comm(comm_target_name, data,
                                               that.callbacks(), data, model_id)
```
with a command saying that only the first one will be kept?

Comment (lbustelo) [inline widgetsnbextension/src/manager.js]: @SylvainCorlay Hmm... not sure I understand the backwards incompatibility wrt jupyter-js-widgets. That code only implements a stub. I do see a problem if the widgetsnbextension and ipywidgets get out of sync.

I'm fine with making the change. How about a target version for removal? Or does it even matter?

Comment (SylvainCorlay) [inline widgetsnbextension/src/manager.js]: hmm, indeed, this change is in the nbextension manager.

**Comment by SylvainCorlay** [comment]
@lbustelo I am 👍 on this change generally, and I would like to push out a 5.0.1 or 5.1.0 pretty quickly with that change so that you guys can iterate.

@jdfreder @jasongrout it is time to say something !

**Comment by jasongrout** [comment]
@SylvainCorlay: I have this to say. Can we *please* merge my 6.x branch to master? :)

This looks okay to me, but I haven't thought a lot about ramifications, nor have I tested it.

**Comment by jdfreder** [comment]
I think this is okay, as long as the information is available on comm open.

**Comment by lbustelo** [comment]
I tested the changes with R and Scala and with this I'm able to continue adding the declarativewidgets support in top of the widgetnbextension by just supporting the new comm target_name.

**Pull request merged**